eppo/src/main/java/cloud/eppo/android/dto/EppoValue.java

```diff
@@ -1,113 +1,110 @@
+        return new EppoValue(boolValue);
     }
 
-    public static EppoValue valueOf(boolean value) {
-        return new EppoValue(Boolean.toString(value), EppoValueType.Boolean);
+    public static EppoValue valueOf(double doubleValue) {
+        return new EppoValue(doubleValue);
     }
 
-    public static EppoValue valueOf(List<String> value) {
-        return  new EppoValue(value);
+    public static EppoValue valueOf(String stringValue) {
+        return new EppoValue(stringValue);
     }
 
-    public static EppoValue valueOf(JsonElement json) {
-        return new EppoValue(json);
+    public static EppoValue valueOf(List<String> value) {
+        return new EppoValue(value);
     }
 
-    public static EppoValue valueOf() {
-        return  new EppoValue(EppoValueType.Null);
+    public boolean booleanValue() {
+        return this.boolValue;
     }
 
     public double doubleValue() {
-        return Double.parseDouble(value);
+        return this.doubleValue;
     }
 
     public String stringValue() {
-        return value;
+        return this.stringValue;
     }
 
-    public boolean boolValue() {
-        return Boolean.valueOf(value);
+    public List<String> stringArrayValue() {
+        return this.stringArrayValue;
     }
 
-    public List<String> arrayValue() {
-        return  array;
+    public boolean isNull() {
+        return type == EppoValueType.NULL;
     }
 
-    public JsonElement jsonValue() {
-        return this.json;
+    public boolean isBoolean() {
+        return this.type == EppoValueType.BOOLEAN;
     }
 
     public boolean isNumeric() {
-        try {
-            Double.parseDouble(value);
-            return true;
-        } catch (Exception e) {
-            return false;
-        }
+        return this.type == EppoValueType.NUMBER;
     }
 
-    public boolean isArray() {
-        return type == EppoValueType.ArrayOfStrings;
+    public boolean isString() {
+        return this.type == EppoValueType.STRING;
     }
 
-    public boolean isBool() {
-        return  type == EppoValueType.Boolean;
-    }
-
-    public boolean isNull() {
-        return type == EppoValueType.Null;
-    }
-
-    public boolean isJSON() {
-        return type == EppoValueType.JSON;
+    public boolean isStringArray() {
+        return type == EppoValueType.ARRAY_OF_STRING;
     }
 
+    @NonNull
     @Override
     public String toString() {
-        return "EppoValue{" +
-                "value='" + value + '\'' +
-                ", type=" + type +
-                ", array=" + array +
-                '}';
+        switch(this.type) {
+            case BOOLEAN:
+                return this.boolValue.toString();
+            case NUMBER:
+                return this.doubleValue.toString();
+            case STRING:
+                return this.stringValue;
+            case ARRAY_OF_STRING:
+                return String.join(" ,", this.stringArrayValue);
+            default: // NULL
```

Comment:
Or just be explicit about it

```suggestion
            case NULL:
```

Reply:
ha true, will do